### PR TITLE
Type annotations for `celery/utils/text.py`

### PR DESCRIPTION
**Pull request by max-muoto**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryq.dev/en/main/contributing.html).

## Description

Type annotations for `celery/utils/text.py` as a part of https://github.com/celery/celery/issues/7394.

### CODE REVIEW

File: celery/utils/text.py [hunk 1]
```diff
@@ -86,33 +78,31 @@ def abbrtask(S, max):
...
     return text
 
 
-def pretty(value, width=80, nl_width=80, sep='\n', **kw):
-    # type: (str, int, int, str, **Any) -> str
+def pretty(value: str, width: int = 80, nl_width: int = 80, sep: str = '\n', **
+           kw: Any) -> str:
     """Format value for printing to console."""
     if isinstance(value, dict):
-        return f'{{{sep} {pformat(value, 4, nl_width)[1:]}'
```
Comment (max-muoto) [inline celery/utils/text.py:115]: Syntax error here.

**Comment by Nusnus** [comment]
There are some issues with the CI so the tests aren't running correctly.
I will handle this soon, FYI

**Comment by max-muoto** [comment]
> There are some issues with the CI so the tests aren't running correctly. I will handle this soon, FYI

Sounds good, thanks!

**Comment by Nusnus** [comment]
The failure is real: https://github.com/celery/celery/actions/runs/4332357772/jobs/7565145570#step:8:1002

**Comment by Nusnus** [comment]
```celery/utils/text.py:50: in <module>
      def join(l: list[str], sep: str = '\n') -> str:
  E   TypeError: 'type' object is not subscriptable```

**Comment by max-muoto** [comment]
> ```
>       def join(l: list[str], sep: str = '\n') -> str:
>   E   TypeError: 'type' object is not subscriptable```
> ```

My bad there, just fixed!

**Comment by Nusnus** [comment]
```ERROR t/unit/bin/test_beat.py - NameError: name 'Match' is not defined
[1068](https://github.com/celery/celery/actions/runs/4332762108/jobs/7565753809#step:8:1069)
  ERROR t/unit/bin/test_worker.py - NameError: name 'Match' is not defined```

**Comment by max-muoto** [comment]
> ```
> ERROR t/unit/bin/test_beat.py - NameError: name 'Match' is not defined
> ERROR t/unit/bin/test_worker.py - NameError: name 'Match' is not defined
> ```

Ah forgot to quote that, as it should be since it's conditionally imported if type checking is enabled. Fixed.

File: celery/utils/text.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
 """Text formatting utilities."""
```
Comment (auvipy) [inline celery/utils/text.py:1]:
```suggestion
"""Text formatting utilities."""
from __future__ import annotations
```

**Review by auvipy — CHANGES_REQUESTED** [review]

**Review by auvipy — APPROVED** [review]

**Pull request merged**